src/ucs/config/parser.c

```diff
@@ -1834,19 +1840,23 @@ static void ucs_config_parser_warn_unused_env_vars(const char *prefix)
     char *saveptr;
     int truncated;
     int ret;
+    ucs_string_buffer_t strb;
 
-    if (!ucs_global_opts.warn_unused_env_vars) {
+    if (!ucs_config_parser_env_vars_track()) {
```

Comment:
we want to warn about unused env vars, even if log level is WARN and not INFO

Reply:
yes, and we have `||` in this func
` return ucs_global_opts.warn_unused_env_vars || ucs_log_is_enabled(UCS_LOG_LEVEL_INFO)`

Reply:
can we move this check to beginning of ucs_config_parser_print_env_vars_once (and also remove from ucs_config_parser_mark_env_var_used and dismantle ucs_config_parser_env_vars_track)

Reply:
I can't remove it from `ucs_config_parser_mark_env_var_used `, because it is also called from `ucs_config_apply_config_vars`.
Also not sure why you meant by `dismantle ucs_config_parser_env_vars_track`

Reply:
ok